apps/app-article-server/src/server/index.js

```diff
@@ -32,87 +35,93 @@ app.get("/article/:id", async (req, res) => {
-      tags={article.tags || []}
-      relatedArticles={article.relatedArticles || []}
-      preamble={article.preamble}
-      articleType={article.articleType}
-      articleTypeDetails={article.articleTypeDetails}
-      publishingTime={article.publishingTime}
-      updateTime={article.updateTime}
-      authors={article.authors}
-      premium={article.premium}
-      isPreview={isPreviewArticle}
-      mostReadArticles={mostReadArticles}
-      fontSize={queryParams.fontSize}
-      darkModeEnabled={queryParams.mode === "dark"}
-      queryString={queryString}
-      paper={paper}
-    />
-  );
+	let article;
+	let isPreviewArticle;
+	// If the response.data is an article already, meaning we can find the article id (uuid) there, great! We have an article
+	if (_.has(articleResponse, "uuid")) {
+	  article = articleResponse;
+	  isPreviewArticle = false;
+	  // Otherwise, it might be a preview article, so we need to dig a bit deeper into the JSON object
+	} else if (_.has(articleResponse, "not_entitled")) {
+	  article = _.get(articleResponse, "not_entitled.articlePreview");
+	  isPreviewArticle = true;
+	}
+	const mostReadArticles = responses[1].data;
+	const user = _.get(responses[2], "data");
 
-  const updatedArticle = _.merge(article, {
-    mostReadArticles: mostReadArticles,
-    isPreview: isPreviewArticle,
-    fontSize: queryParams.fontSize,
-    darkModeEnabled: queryParams.mode === "dark",
-    queryString: queryString,
-    paper: paper,
-  });
+	const articleJSX = (
+	  <Article
+	    title={article.title}
+	    mainImage={article.mainImage}
+	    body={article.body}
+	    tags={article.tags || []}
+	    relatedArticles={article.relatedArticles || []}
+	    preamble={article.preamble}
+	    articleType={article.articleType}
+	    articleTypeDetails={article.articleTypeDetails}
+	    publishingTime={article.publishingTime}
+	    updateTime={article.updateTime}
+	    authors={article.authors}
+	    premium={article.premium}
+	    isPreview={isPreviewArticle}
+	    mostReadArticles={mostReadArticles}
+	    fontSize={queryParams.fontSize}
+	    darkModeEnabled={queryParams.mode === "dark"}
+	    queryString={queryString}
+	    paper={paper}
+	  />
+	);
 
-  const markup = ReactDOM.renderToString(articleJSX);
-  const html = generateHtml(markup, updatedArticle, user);
-  res.send(html);
+	const updatedArticle = _.merge(article, {
+	  mostReadArticles: mostReadArticles,
+	  isPreview: isPreviewArticle,
+	  fontSize: queryParams.fontSize,
+	  darkModeEnabled: queryParams.mode === "dark",
+	  queryString: queryString,
+	  paper: paper,
+	});
+	const markup = ReactDOM.renderToString(articleJSX);
+	const finalHtml = generateHtml(markup, updatedArticle, user);
+	res.send(finalHtml);
+      })
+    )
+    .catch((errors) => {
+      const markup = ReactDOM.renderToString(<ErrorPage message={"Artikeln kunde inte hämtas!"} />);
+      const html = generateHtml(markup);
+      console.warn("Failed to fetch article!", {
```

Comment:
Does this `console.warn` need removing?

Reply:
For javascript running in the browser, one might argue logging to the console is indeed unnecessary in production.

In this case however, the logging is actually useful, as they are saved to google cloud logs.